x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1006 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { CredentialsLogic } from './credentials_logic';
+import { ADMIN, PRIVATE } from '../../constants/credentials';
+
+jest.mock('../../../shared/http', () => ({
+  HttpLogic: { values: { http: { get: jest.fn(), delete: jest.fn() } } },
+}));
+import { HttpLogic } from '../../../shared/http';
+
+function mount(defaults?) {
```

Comment:
This mount helper just makes the tests a bit more readable. It handles calling `resetContext` and `CredentialsLogic.mount`, while also giving us a convenient API for initializing our logic values with particular data.

```
        mount({
          activeApiTokenRawName: 'foo',
        });
```

That would initialize the credentials logic store with `activeApiTokenRawName` set to `foo`, rather than the default value it would normally receive upon initialization.

I think this would be a nice pattern to carry through to other logic tests as well.

Reply:
Agreed - super cool pattern! 🎉